c7n/resources/route53.py

```diff
@@ -485,3 +486,76 @@ def process(self, resources, event=None):
             elif not logging and not state:
                 results.append(r)
         return results
+
+
+@resources.register('resolver-qlc')
+class ResolverQueryLogConfig(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'route53resolver'
+        arn_type = 'query_log_config'
+        enum_spec = ('list_resolver_query_log_configs', 'ResolverQueryLogConfigs', None)
+        name = 'Name'
+        id = 'Id'
+        cfn_type = 'AWS::Route53Resolver::ResolverQueryLoggingConfig'
+
+    permissions = (
+        'route53resolver:ListResolverQueryLogConfigs',
+        'route53resolver:ListResolverQueryLogConfigAssociations')
+
+    def augment(self, rqlcs):
+        client = local_session(self.session_factory).client('route53resolver')
+        for rqlc in rqlcs:
+            rqlc['Tags'] = self.retry(
+                client.list_tags_for_resource,
+                ResourceArn=rqlc['Arn'])['Tags']
+        return rqlcs
+
+
+@ResolverQueryLogConfig.action_registry.register('associate-vpc')
+class ResolverQueryLogConfigAssociate(BaseAction):
+    """Associates ResolverQueryLogConfig to a VPC
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: r53-resolver-query-log-config-associate
+            resource: resolver-qlc
+            filters:
+              - type: value
+                key: 'Name'
+                op: eq
+                value: "Test-rqlc"
+            actions:
+              - type: associate-vpc
```

Comment:
I think we'll want a filter for detecting query log config to vpc configuration as well, we don't want the policy to match on query logs, if they are already associated.

Reply:
@kapilt  many thanks for the great suggestion .. seeking a little more clarification, how do you reckon it implemented ... a separate filter  or  augmenting the resource data with association data and then using the value filter in policy.

Reply:
@kapilt added a new filter .. not sure if this what you had envisioned.